sbi/utils/user_input_checks.py

```diff
@@ -569,6 +572,10 @@ def process_x(
 
     x = atleast_2d(torch.as_tensor(x, dtype=float32))
 
+    # If x_shape is provided, we can fix a missing batch dim for >1D data.
+    if x_shape is not None and len(x_shape) > len(x.shape):
+        x = x.unsqueeze(0)
```

Comment:
Hm, I'm confused:

During training, `training_x.shape = (N, T, D)`. Thus, the inferred `x_shape == (T, D)`.

Then, here, `x.shape == (T, D)` if no batch dimension is passed (but iid x). So this if-case would not unsqueeze. Is this a bug or am I missing something?

Reply:
Also, doesn't this fix come too late? The error has already been raised above?

Reply:
in most cases we have the `x_shape` inferred from the training data and stored in `self._x_shape`, including the batch. E.g., for SNPE with 10-trial 2D data `x_shape=(1, 10, 2)`. If the incoming `x.shape` does not have a batch dimension, e.g., `x.shape = (10, 2)` then we can fix that here.

Reply:
the error occurs only for SNPE in line 581 because it requires a singleton batch dimension.

Reply:
Ah, ok I had not known that `self._x_shape` contains the batch dim. Makes sense.

Regarding the second point: sorry, I got confused by github's editor and thought that the fixes above were in the same function...makes sense now.

Reply:
I adapted the `process_x` tests to make sure this is working for all combinations of inputs.

Reply:
Thanks! Good to go!